File: test-suite-kotlin/src/test/kotlin/io/micronaut/tracing/docs/HelloService.kt
package io.micronaut.tracing.docs

// tag::imports[]
import io.micronaut.tracing.annotation.ContinueSpan
import io.micronaut.tracing.annotation.NewSpan
import io.micronaut.tracing.annotation.SpanTag
import jakarta.inject.Singleton
// end::imports[]

// tag::clazz[]
@Singleton
open class HelloService {

    @NewSpan("hello-world") // <1>
    open fun hello(@SpanTag("person.name") name: String): String { // <2>
        return greet("Hello $name")
    }

    @ContinueSpan // <3>
    open fun greet(@SpanTag("hello.greeting") greet: String): String {
        return greet
    }
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/tracing/docs/HelloService.py
# tag::imports[]
from typing import Annotated

from jakarta.inject import Singleton
from micronaut.tracing.annotation import ContinueSpan, NewSpan, SpanTag
# end::imports[]


# tag::clazz[]
@Singleton
class HelloService:

    @NewSpan("hello-world")  # <1>
    def hello(self, name: Annotated[str, SpanTag("person.name")]) -> str:  # <2>
        return self.greet("Hello " + name)

    @ContinueSpan  # <3>
    def greet(self, greet: Annotated[str, SpanTag("hello.greeting")]) -> str:
        return greet
# end::clazz[]
